.github/workflows/upload_coverage_report.yml

```diff
@@ -23,13 +20,9 @@ jobs:
 
       - name: Upload coverage report
         if: ${{ hashFiles('build/coverage_report.xml') != '' }}
-        uses: wandalen/wretry.action@ffdd254f4eaf1562b8a2c66aeaa37f1ff2231179 #v3.7.3
```

Comment:
There was a note that this extension was a bit dodgy, so let's remove it
I'm updating codecov/codecov-action hoping it is more stable now and we don't need action retry at all.